### PR TITLE
Add preliminary framework for generating X3D (and eventually RTT) mesh files.

### PR DESCRIPTION
**Pull request by RyanWollaeger**
### Background

* Some mesh file generation capability will facilitate testing Draco's supported mesh parsers.

### Purpose of Pull Request

* Add an orthogonal-structured class mesh type for testing.
* Add an X3D mesh file generator, agnostic of mesh structure or dimension.
* [Concerns Redmine Issue #2466](https://rtt.lanl.gov/redmine/issues/2466)

### Description of changes

+ Add **src/mesh/python** subdirectory.
+ Add **mesh_type** file intended to contain python mesh classes.
+ Add `orth_2d_mesh` class to mesh_types: orthogonal quad 2D data.
+ Add **x3d_generator**, which creates X3D files from mesh objects.
+ Import mesh_types into x3d_generator (for the mesh objects).
+ Add `orth_3d_mesh` class stub for eventual 3D mesh file generation.

Notes:
- An equivalent **rtt_generator** can be added to take mesh object data
  to the RTT mesh file format.
- Consider using Python 3.7 **dataclasses** for mesh classes.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by RyanWollaeger** [comment]
@brryan Maybe the Voronoi mesh can be introduced as a class with the relevant data in this new python subdirectory, so we can then add an rtt_generator.py (following x3d_generator.py) which can output the format.  Let me know if this seems off-base.

File: src/mesh/python/mesh_types.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+# -------------------------------------------*-python-*------------------------------------------- #
```
Comment (KineticTheory) [inline src/mesh/python/mesh_types.py:2]: Please add standard copyright block. to both python files.

**Review by KineticTheory — APPROVED** [review]

File: src/mesh/python/mesh_types.py (hunk 1 above)
Comment (RyanWollaeger) [reply src/mesh/python/mesh_types.py:2]: Ah - thank you for catching this - will do.

Comment (RyanWollaeger) [reply src/mesh/python/mesh_types.py:2]: Just added these fields - thank you!

**Comment by KineticTheory** [comment]
I'm ready to merge this PR.  @brryan @kgbudge Do you have any comments for @RyanWollaeger?

**Pull request merged**